lineedit/editing.c

```diff
@@ -2974,7 +2983,7 @@ void cmd_vi_edit_and_accept(wchar_t c __attribute__((unused)))
         fwprintf(f, L"%ls\n", le_main_buffer.contents);
         fclose(f);
 
-        wchar_t *command = malloc_wprintf(L"vi %s", tempfile);
+        wchar_t *command = malloc_wprintf(L"%s %s", editor,  tempfile);
```

Comment:
I think we can simply do like this

```suggestion
        wchar_t *command = malloc_wprintf(L"${VISUAL:-${EDITOR:-vi}} %s", tempfile);
```

The `fc` built-in invokes an editor in a similar way: https://github.com/magicant/yash/blob/97f4858970e7cbf790452a7b3d578ecb75a8ec96/history.c#L1604-L1605